src/raft_log.rs

```diff
@@ -354,12 +354,24 @@ impl<T: Storage> RaftLog<T> {
         term > self.last_term() || (term == self.last_term() && last_index >= self.last_index())
     }
 
-    /// Returns any entries since the a particular index.
-    pub fn next_entries_since(&self, since_idx: u64) -> Option<Vec<Entry>> {
-        let offset = cmp::max(since_idx + 1, self.first_index());
-        let committed = self.committed;
-        if committed + 1 > offset {
-            match self.slice(offset, committed + 1, None) {
+    /// Returns any entries between the `since_idx`(if none, use applied)
+    /// and the min(`persisted_idx`(if any), commit index) + 1.
+    pub fn next_entries_between(
+        &self,
+        since_idx: Option<u64>,
+        persisted_idx: Option<u64>,
```

Comment:
I suggest to use `u64` instead. So the match is unnecessary.

Reply:
The function `next_entries` still needs this.

Reply:
`next_entries` can call `self.next_entries_between(self.applied, self.committed + 1)`.

Reply:
Got it.